pjsip/src/pjsip/sip_uri.c

```diff
@@ -256,10 +256,11 @@ static pj_ssize_t pjsip_url_print(  pjsip_uri_context_e context,
 {
     int printed;
     char *startbuf = buf;
-    char *endbuf = buf+size;
+    char *endbuf = buf+size-1; // Need to minus one for NULL terminator
     const pj_str_t *scheme;
     const pjsip_parser_const_t *pc = pjsip_parser_const();
 
+    if (endbuf < startbuf) return -1;
```

Comment:
You're checking here for the special case of size == 0, which is fine, but you're not doing this check in all the other places where you do
`char *endbuf = buf+size-1;`
like for example here:
https://github.com/pjsip/pjproject/blob/4813f17bff2e95780403276d4cfd88785c187a0e/pjsip/src/pjsip/sip_auth_msg.c#L128

https://github.com/pjsip/pjproject/blob/4813f17bff2e95780403276d4cfd88785c187a0e/pjsip/src/pjsip/sip_auth_msg.c#L294

https://github.com/pjsip/pjproject/blob/4813f17bff2e95780403276d4cfd88785c187a0e/pjsip/src/pjsip/sip_tel_uri.c#L185

https://github.com/pjsip/pjproject/blob/4813f17bff2e95780403276d4cfd88785c187a0e/pjsip/src/pjsip/sip_uri.c#L562

Do you not need to add this check in these places as well?

Reply:
I thought of that, and the reason I didn't put it in other places is because it will be checked in `copy_advance()`.